### PR TITLE
Make WorkspaceModel.Groups, Notes IEnumerable's

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link To YouTrack:
[MAGN-7862](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7862) Make WorkspaceModel.Groups, Notes IEnumerable's

I changed `ObservableCollection`'s to `IEnumerable`'s. In task it is said, that I should change `Note` and `Group` collections, but I didn't find `Group` collection. I assume that Groups are Annotations.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@lukechurch 

### FYIs

@pboyer

### CODE REVIEW

File: src/DynamoCore/Models/WorkspaceModel.cs [hunk 1]
```diff
@@ -62,9 +62,9 @@ internal int CurrentPasteOffset
         private string author = "None provided";
         private string description;
         private bool hasUnsavedChanges;
-        private readonly ObservableCollection<NodeModel> nodes;
```
Comment (aosyatnik) [inline src/DynamoCore/Models/WorkspaceModel.cs]: Do we need `ObservableCollection` here? I thought, that `List` is faster, isn't it?

File: src/DynamoCore/Models/WorkspaceModel.cs [hunk 2]
```diff
@@ -743,11 +838,59 @@ public NoteModel AddNote(bool centerNote, double xPos, double yPos, string text,
...
+        private void RemoveNote(NoteModel note)
+        {
+            lock (notes)
+            {
+                if (!notes.Remove(note)) return;
+            }
+            OnNoteRemoved(note);
+        }
+
+        private void AddNewAnnotation(AnnotationModel annotation)
```
Comment (aosyatnik) [inline src/DynamoCore/Models/WorkspaceModel.cs]: I wanted to name it like
```c#
private void AddAnnotation(AnnotationModel annotation)
```
but there is public method with the same name and the same incoming parameter. (line 889)

File: src/DynamoCore/Models/WorkspaceModel.cs (hunk 1 above)
Comment (pboyer) [inline src/DynamoCore/Models/WorkspaceModel.cs]: Yes, this is right.

**Comment by pboyer** [comment]
@aosyatnik I'd recommend ensuring that both DynamoRevit and Reach continue to build after making this change. Let me know if you need any help with that.

**Comment by aosyatnik** [comment]
Hi, @pboyer , Revit and Reach builds. I run tests. 9520 passed and 1 failed. I assume, that the failing test 
is `TestWarningMessageLog`, that doesn't work now.
image

**Comment by pboyer** [comment]
@aosyatnik I'm going to bring this in on Monday to prevent destabilizing the build.

**Pull request merged**